include/small/ibuf.h

```diff
@@ -144,6 +144,10 @@ ibuf_alloc(struct ibuf *ibuf, size_t size)
 	return ptr;
 }
 
+/** Expand the buffer, or shrink it to return the memory. */
+void *
+ibuf_resize(struct ibuf *ibuf, size_t new_size);
```

Comment:
I don't quite like the semantics of this method, because it's rather difficult to comprehend:
- Does it set the capacity of the buffer to `new_size`? Not exactly: the new capacity can't be less than the size of data stored in the buffer.
- Okay, does it at least set the capacity of the buffer to `new_size` *in case* the size of data stored in the buffer is less than `new_size`? Again, not exactly: the new capacity may actually be greater, because it's computed from the start capacity.

Maybe, just drop the argument and rename this method to `ibuf_shrink`? The method would shrink the capacity of the buffer to the minimal capacity needed to store the data written to the buffer so far, but never less than the start capacity. Do we really need the ability to specify the target size?

This question is worth discussing with @alyapunov.

Reply:
Done.